modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/global_search/GlobalSearch.js

```diff
@@ -0,0 +1,408 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayDropDown from '@clayui/drop-down';
+import {ClayInput} from '@clayui/form';
+import {ClayIconSpriteContext} from '@clayui/icon';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClaySticker from '@clayui/sticker';
+import {useIsMounted} from 'frontend-js-react-web';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+import ServiceProvider from '../../ServiceProvider/index';
+import {debouncePromise} from '../../utilities/debounce';
+import {formatActionUrl, getRandomId} from '../../utilities/index';
+import {showErrorNotification} from '../../utilities/notifications';
+
+const OrderResource = ServiceProvider.AdminOrderAPI('v1');
+const AccountsResource = ServiceProvider.AdminAccountAPI('v1');
+const CatalogResource = ServiceProvider.DeliveryCatalogAPI('v1');
+
+function SearchItem({className, href, label, stickerShape, thumbnailUrl}) {
+	return (
+		<ClayDropDown.Item className={className} href={href}>
+			<ClaySticker className="mr-2" shape={stickerShape}>
+				<ClaySticker.Image alt={label} src={thumbnailUrl} />
+			</ClaySticker>
+			{label}
+		</ClayDropDown.Item>
+	);
+}
+
+const composeDataUpdate = (
+	promise,
+	isMounted,
+	updateData,
+	updateCounter,
+	updateLoader
+) => {
+	return promise
+		.then((res) => {
+			if (!isMounted()) {
+				return;
+			}
+			updateData(res.items);
+			updateCounter(res.totalCount);
+		})
+		.catch((e) => {
+			showErrorNotification(e.message);
+			updateData(null);
+			updateCounter(null);
+		})
+		.finally(() => {
+			if (!isMounted()) {
+				return;
+			}
+			updateLoader(false);
+		});
+};
+
+function GlobalSearch(props) {
+	const [query, updateQuery] = useState('');
+	const isMounted = useIsMounted();
+	const inputRef = useRef(null);
+	const dropdownRef = useRef(null);
+	const [accountsLoading, updateAccountsLoading] = useState(false);
+	const [accounts, updateAccounts] = useState(null);
+	const [accountsCount, updateAccountsCount] = useState(null);
+	const [debouncedGetAccounts, updateDebouncedGetAccounts] = useState(null);
+	const [ordersLoading, updateOrdersLoading] = useState(false);
+	const [orders, updateOrders] = useState(null);
+	const [ordersCount, updateOrdersCount] = useState(null);
+	const [debouncedGetOrders, updateDebouncedGetOrders] = useState(null);
+	const [productsLoading, updateProductsLoading] = useState(false);
+	const [products, updateProducts] = useState(null);
+	const [productsCount, updateProductsCount] = useState(null);
+	const [debouncedGetProducts, updateDebouncedGetProducts] = useState(null);
+	const [active, setActive] = useState(false);
+	const [ids] = useState({
+		input: 'global-search-input' + getRandomId(),
+		menu: 'global-search-menu' + getRandomId(),
```

Comment:
Is this search truly "global"? If so, you could avoid recomputing random IDs on every render and just do it once outside the component. (I guess though, it's not global, or you would probably have done that...)